guides/source/association_basics.md

```diff
@@ -1713,10 +1743,41 @@ record's `user_id` value will be the `guid` value of `@user`.
 
 The `:source` option specifies the source association name for a `has_many :through` association. You only need to use this option if the name of the source association cannot be automatically inferred from the association name.
 
+```ruby
+class Assembly < ApplicationRecord
+  has_many :manifests
+  has_many :components, through: :manifests, source: :part
+end
+
+class Manifest < ApplicationRecord
+  belongs_to :assembly
+  belongs_to :part
+end
+
+class Part < ApplicationRecord
+  has_many :manifests
+  has_many :assemblies, through: :manifests
+end
+```
+
```

Comment:
Same here, we can combine the source/type examples into one and just show below source_type